text/0000-from-elem-with-love.md

```diff
@@ -0,0 +1,73 @@
+- Feature Name: from_elem_with_love
+- Start Date: 2015-02-11
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add back `Vec::from_elem`.
+
+# Motivation
+
+High demand, mostly. There are currently a few ways to achieve the behaviour of `Vec::from_elem(elem, n)`:
+
+```
+// #1
+let vec = Vec::new();
+for i in range(0, n) {
+    vec.push(elem.clone())
+}
+```
+
+```
+// #2
+let vec = vec![elem; n]
+```
+
+```
+// #3
+let vec = Vec::new();
+vec.resize(elem, n);
+```
+
+```
+// #4
+let vec: Vec<_> = (0..n).map(|_| elem.clone()).collect()
+```
+
+```
+// #5
+let vec: Vec<_> = iter::repeat(elem).take(n).collect();
+```
+
+None of these quite match the convenience, power, and performance of:
+
+```
+let vec = Vec::from_elem(elem, n)
+```
+
+* `#1` is verbose *and* slow, because each `push` requires a capacity check
+* `#2` only works for a Copy `elem` and const `n`
+* `#3` needs a temporary, but should be otherwise identical performance-wise.
+* `#4` and `#5` suffer from the untrusted iterator len problem performance wise (which is only a temporary 
+argument, this will be solved sooner rather than later), and are otherwise verbose and noisy. They also 
```

Comment:
To me personally the 5th option is indeed the idiomatic method by which I would recommend creating a `Vec` today. Opinions of verboseness/noisiness aside, it would be useful to quantify precisely why this this is slow beyond the trusted iterator length problem, as well as backing this claim up with numbers.

I've created a [sample benchmark](https://gist.github.com/alexcrichton/6a16485d26b4c9a46d1e) which *today* resolves the performance argument for options 4/5 (at least for this one use case). This is boiling down to me trying to say that performance is a fixable problem with the exact APIs we have today (especially with trusted iterator lengths) and it may not be a strong enough motivation for this change.